### PR TITLE
Create Issue Templates

### PR DESCRIPTION
**Pull request by ax3l**
Create issue templates for:
- [x] bugs
- [x] installation issues
- [x] feature requests
- [x] blank
- [ ] usage question -> link to [Discussions](https://github.com/ECP-WarpX/WarpX/discussions)

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/feature_request.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+A clear and concise description of the context.
+
+**What feature would you like to add to WarpX?**
+A clear and concise description of the feature.
+
+**What alternatives have you considered?**
+A clear and concise description of any alternative solutions or features you have considered.
+
+**Additional information**
+If applicable, please add other info about the feature request.
```
Comment (roelof-groenewald) [inline .github/ISSUE_TEMPLATE/feature_request.md]: Maybe here or in the "what feature" section above we should ask for a publication reference if the new feature is a model found in other codes?

**Review by roelof-groenewald — COMMENTED** [review]
Looks good to me! I think this will help reduce the initial back-and-forth from new issues where we usually just ask about the input script, machine setup, etc.

Just one suggestion for a possible improvement.

File: .github/ISSUE_TEMPLATE/feature_request.md (hunk 1 above)
Comment (EZoni) [reply .github/ISSUE_TEMPLATE/feature_request.md]:
```suggestion
**Additional information**
If applicable, please add other info about the feature request (e.g., links to existing codes, publications, etc.).
```

**Review by roelof-groenewald — APPROVED** [review]
Thanks for the update!
I guess the "usage question" template is still not made. Maybe that can be done in a follow-up PR so we can see how well these other template work in the meantime?

**Comment by EZoni** [comment]
@roelof-groenewald 

Thanks for your feedback, yes, I agree, this will be for sure an iterative process. The discussions page is already quite active and we can see how well this first draft works with the next new issues coming in, while refining things (like the usage question link) in follow-up PRs.

I would still wait for more feedback on the current templates from other developers before merging, though, just to get as much diverse inputs and perspectives as we can.

**Review by dpgrote — APPROVED** [review]
This is great! Thanks! It is already very thorough.
Is there a need for a template for pull requests (for example like with AMReX)?

**Comment by EZoni** [comment]
@dpgrote 

Thanks! Yes, I think we could add PR templates as well, but we felt issues were more urgent since they are the more user-facing of the two. That's the only reason why we started with issues here.

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+**What behavior did you expect?**
+A clear and concise description of what you expected to happen.
+
+**Tell us about your software environment**
+- operating system: [name and version]
+- version of WarpX: [e.g., latest, 24.09, etc.]
+- installed WarpX via: [e.g., conda-forge, spack, pip, brew, from source, module system, etc.]
+- HPC machine: [Are you running on an HPC cluster? We might support it already: https://warpx.readthedocs.io/en/latest/install/hpc.html]
+- other relevant dependency names and versions
+
```
Comment (aeriforme) [inline .github/ISSUE_TEMPLATE/bug_report.md:54]: Should we add something like:

Have you tried compiling and running in debug mode?
If so, attach one of the Backtraces files. 

I feel like this is also a recurring step that could be skipped by adding it here.

Comment (atmyers) [inline .github/ISSUE_TEMPLATE/bug_report.md:54]:
```suggestion

**How did you run**
Tell us a bit about how you were running WarpX when you encountered the bug. 
-One MPI rank or several? 
-With or without OpenMP?
-CPUs or GPUs? 
-What kind of GPU?
-Etc.
```

**Comment by EZoni** [comment]
Thanks all for the feedback! I'll add your input asap. I wonder if we should rely on a check box approach if the amount of info starts to become large (maybe people find it more appealing to check boxes rather than spelling out info). I'll see if I can come up with something along these lines, for some of the info that we request, and ping you for review.

File: .github/ISSUE_TEMPLATE/feature_request.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
+---
+name: Feature request
+about: Request a new feature
+title: ''
+labels: [enhancement]
+assignees: ''
+---
```
Comment (EZoni) [inline .github/ISSUE_TEMPLATE/feature_request.md]:
```suggestion
---
name: Feature request
about: Suggest a new feature or enhancement.
labels: [enhancement]
---
```

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+Please minimize your inputs/scripts to be concise and focused on the issue.
+For instance, make the simulation scripts as small and fast to run as possible.
+
+## Error messages
+Please paste any relevant error messages or console output.
+
+## System information
+Please check all relevant boxes and provide details.
+
+- Operating System (name and version):
```
Comment (EZoni) [inline .github/ISSUE_TEMPLATE/bug_report.md]:
```suggestion
- Operating system (name and version):
```

File: .github/ISSUE_TEMPLATE/installation-issue.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
+---
+name: Installation issue
+about: Cannot install WarpX? Let us know.
+title: ''
+labels: [install, question]
+assignees: ''
+---
```
Comment (EZoni) [inline .github/ISSUE_TEMPLATE/installation-issue.md]:
```suggestion
---
name: Installation issue
about: Report an issue with installing or setting up WarpX
labels: [install]
---
```

File: .github/ISSUE_TEMPLATE/feature_request.md [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+name: Feature request
+about: Suggest a new feature or enhancement.
+labels: [enhancement]
+---
+
```
Comment (EZoni) [inline .github/ISSUE_TEMPLATE/feature_request.md:6]:
```suggestion

_Please remove any sensitive information (e.g., passwords, API keys) from your submission. Thank you for taking the time to report this issue. We will respond as soon as possible._

```

File: .github/ISSUE_TEMPLATE/installation-issue.md [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+name: Installation issue
+about: Report an issue with installing or setting up WarpX
+labels: [install]
+---
+
```
Comment (EZoni) [inline .github/ISSUE_TEMPLATE/installation-issue.md:6]:
```suggestion

_Please remove any sensitive information (e.g., passwords, API keys) from your submission. Thank you for taking the time to report this issue. We will respond as soon as possible._

```

**Comment by EZoni** [comment]
I did a bit of work on these templates, trying to make them more uniform, so feel free to re-review if you'd like. There's a bit of repetition between the bug report and the installation issue, to be decided if we want to keep them separate or merge them.

**Review by atmyers — APPROVED** [review]

**Pull request merged**